src/app/view/components/login/dot-login-component/dot-login.component.ts

```diff
@@ -0,0 +1,181 @@
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+
+@Component({
+    selector: 'dot-login-component',
+    templateUrl: './dot-login.component.html',
+    styleUrls: ['./dot-login.component.scss']
+})
+/**
+ * The login component allows the user to fill all
+ * the info required to log in the dotCMS angular backend
+ */
+export class DotLoginComponent implements AfterViewInit, OnInit {
+    @Input() message = '';
+    @Input() passwordChanged = false;
+    @Input() resetEmailSent = false;
+    @Input() resetEmail = '';
+    @Input()
+    set isLoginInProgress(value: boolean) {
+        this.disableFormFields(value);
+    }
+    @Output() recoverPassword = new EventEmitter<any>();
+    @Output() login = new EventEmitter<DotLoginData>();
+
+    @ViewChild('emailInput') emailInput: ElementRef;
+
+    loginForm: FormGroup;
+    languages: SelectItem[] = [];
+    dotSystemInformation: DotSystemInformation;
+
+    dataI18n: { [key: string]: string } = {};
+
+    isCommunityLicense = true;
+    userIdOrEmailLabel = '';
+    emailAddressLabel = '';
+
+    private i18nMessages: Array<string> = [
+        'email-address',
+        'user-id',
+        'password',
+        'remember-me',
+        'sign-in',
+        'get-new-password',
+        'cancel',
+        'Server',
+        'error.form.mandatory',
+        'angular.login.component.community.licence.message',
+        'reset-password-success',
+        'a-new-password-has-been-sent-to-x',
+        'welcome-back'
+    ];
+
+    constructor(
+        private loginService: LoginService,
+        private loggerService: LoggerService,
+        private fb: FormBuilder
+    ) {}
+
+    ngOnInit() {
+        this.loginService.setAuth({
+            loginAsUser: null,
+            user: null,
+            isLoginAs: false
+        });
+        this.loginForm = this.fb.group({
+            login: ['', [Validators.required]],
+            language: [''],
+            password: ['', [Validators.required]],
+            rememberMe: false
+        });
+        this.renderPageData();
+    }
+
+    ngAfterViewInit(): void {
+        this.emailInput.nativeElement.focus();
+    }
+
+    /**
+     *  Executes the logIn service
+     */
+    logInUser(): void {
+        if (this.loginForm.valid) {
+            this.login.emit(this.loginForm.value);
+        }
+    }
+
+    /**
+     * Execute the change language service
+     */
+    changeLanguage(lang: string): void {
+        this.loginForm.get('language').setValue(lang);
+        this.renderPageData();
+    }
+
+    /**
+     * Display the forgot password card
+     */
+    showForgotPassword(): void {
+        this.recoverPassword.emit();
+    }
+
+    /**
+     * Renders all the labels, images, and placeholder values for the Log In page.
+     */
+    private renderPageData(): void {
+        this.loginService
+            .getLoginFormInfo(this.loginForm.get('language').value, this.i18nMessages)
+            .subscribe(
```

Comment:
We're not getting unsubscribe here, but I think in this case is better if we use async pipe for this, we can map all the properties we have here into an object and use in the template kinda like this:

<img width="468" alt="Screen Shot 2019-03-19 at 9 07 11 AM" src="https://user-images.githubusercontent.com/751424/54617161-6aa63180-4a26-11e9-95cb-d8cb581fa450.png">